src/main/webapp/search-script.js

```diff
@@ -0,0 +1,84 @@
+function countSearchUnits() {
```

Comment:
If you're populating an array, then you're not counting right? 
Renaming to something like collectUnitsFromSearch might make more sense.